src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -176,18 +194,16 @@ let pp_ctor ppf (p : Locations.typed_prog_num) =
       | ls -> Some ls )
     | _ -> None
   in
-  let rec pp_statement_zeroing ppf s =
-    let pp_statement_list_zeroing ppf ls =
-      (list ~sep:cut pp_statement_zeroing) ppf ls
+  let pp_statement_zeroing ppf ls =
```

Comment:
Could you explain the intention of this code (maybe in a comment)? I am a bit confused about what pp_statement_zeroing is supposed to do and how it should be different from pp_statement.
I guess it should not print any code for declarations and it should only generate the code for setting the sizes of declarations and otherwise it's the same as pp_statement. Where do you use this? (Just out of my own interest in understanding the code gen)

Reply:
Yeah, it probably needs a better name now. It basically just elides the declarations at the top-level. I'll change it to something...

Reply:
Ah right. And should it not recurse into SList then? What if we have a bunch of SLists at the toplevel?

Reply:
I'm operating with the assumption that we're still going to be removing SList - I don't know what the semantics around it are at this point.